gbus/tx/mysql/txoutbox.go

```diff
@@ -124,16 +132,34 @@ func NewOutbox(svcName string, txProv gbus.TxProvider, purgeOnStartup bool) *TxO
 		txProv:         txProv,
 		purgeOnStartup: purgeOnStartup,
 		ID:             xid.New().String(),
-		ack:            make(chan uint64, 10000),
-		nack:           make(chan uint64, 10000),
+		ack:            make(chan uint64, 1000000),
+		nack:           make(chan uint64, 1000000),
 		exit:           make(chan bool)}
 	return txo
 }
 
+func (outbox *TxOutbox) ackRec() {
+	for {
+		select {
+		case <-outbox.exit:
+			return
+		case ack := <-outbox.ack:
+			outbox.log().WithField("channel_len", len(outbox.ack)).Debug("length of ack channel")
+			if err := outbox.updateAckedRecord(ack); err != nil {
+				outbox.log().WithError(err).WithField("delivery_tag", ack).Error("failed to update delivery tag")
+			}
+		case nack := <-outbox.nack:
+			outbox.log().WithField("deliver_tag", nack).Info("nack received for delivery tag")
```

Comment:
You don't do Nack anymore?

Reply:
These nacks are the broker sending a negative confirm for the publication so we do not handle them and let the transactional outbox re publish the message the next time it scans for messages that were sent but not yet confirmed (that is done with the scavenging channel/goroutine)